packages/toolpad-core/src/SignInPage/SignInPage.tsx

```diff
@@ -366,7 +364,7 @@ function SignInPage(props: SignInPageProps) {
                         const oauthResponse = await signIn?.(provider, undefined, callbackUrl);
                         setFormStatus((prev) => ({
                           ...prev,
-                          loading: oauthResponse?.error || docs ? false : prev.loading,
+                          loading: false,
```

Comment:
Shouldn't we keep the `oauthResponse?.error` logic here? If not should we just remove the loading and depend on whether it is defined in the `prev`?

Reply:
The idea in this code was to use `oauthResponse?.error` being defined as a way to check that the operation is complete and set the loading state to false (but create an escape hatch. for the docs since no oauthResponse is created inside the docs). 

`prev.loading` will be true in all cases, and letting `loading` stay that way was done so that even if a response is received, the button would remain in a "loading" state till the OAuth redirect occurs, to visually indicate to the user that something is still "pending". This behaviour is present only in the oauth case, since none of the other auth methods involve an intermediate redirect. 

On second thought, I feel it's a nice detail to let the user know that the redirect is pending, so what I feel we can do is to simply remove the additional `docs` check here, and change the oauth docs demos to return a mock error with "this is a fake error message", so that the button goes back into the default state.

<table>
  <thead>
    <tr>
      <th>Current</th>
      <th>After this PR</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <video src="https://github.com/user-attachments/assets/36b0358b-a479-473c-b951-5386bbdc0aea" controls></video>
      </td>
      <td>
        <video src="https://github.com/user-attachments/assets/04efa460-b3c3-47e7-8ae3-fd6b504a84b0" controls></video>
      </td>
    </tr>
  </tbody>
</table>

Reply:
Yeah, that makes sense 👌